### ISSUE TITLE
Windows build issue: CUDPP

### ISSUE THREAD

**Issue by revillo** [NVIDIA/gvdb-voxels#14]
First of all thanks for updating gvdb, really appreciate the support for multiple contexts!
Just wanted to point out a small issue during the build process. The readme states:

"Install CUDA Toolkit 8.0"

However, all the provided CUDPP binaries are compiled for CUDA 9 and are not compatible during a CUDA 8 build.

It was a simple matter of compiling CUDPP again for CUDA 8 and renaming the binaries using the convention:
cudpp_1900x64.lib -> cudpp_1900cu8x64.lib
etc.

**Comment by JC-Larkey** [comment]
How did you do this?

**Comment by inhahe** [comment]
How do you specify which CUDA version is used when building cudpp? Thanks.

**Comment by NeilBickford-NV** [comment]
I think GVDB no longer uses CUDPP as of version 1.1.1 - sorry about that, I need to clean up some of these old issues! But maybe I can help if this is for a different project.

**Comment by inhahe** [comment]
Thank you. Not sure what you mean by a different project. I want to make a
new (simple) project using the GVDB library.

I found a similar issue here
https://forums.developer.nvidia.com/t/problems-with-shared-cudpp/59905, but
I think his circumstances are a little different and I'm not sure which if
any of the solutions apply to me or how to adapt them to my situation.

**Comment by NeilBickford-NV** [comment]
Thanks, that's good to know! I think that thread's a bit out of date now - we removed CUDPP from GVDB and refreshed the build system in January of this year (we now use Thrust instead), so hopefully you shouldn't have to deal with building or linking CUDPP now.

Probably one of the easiest ways to get started making a new project using the GVDB library is to start by modifying one of the examples! If you want to create an app using GVDB outside of the example framework, you can build the `INSTALL` target for GVDB (see the main page for instructions on how to do this) to generate a folder with shared libraries, include files, and a linker library, and then link them into your application as usual. If you're using CMake, you can also try uncommenting https://github.com/NVIDIA/gvdb-voxels/blob/master/source/gvdb_library/CMakeLists.txt#L311-L312 - this should make it so that you can link GVDB using find_package(gvdb), though note that this hasn't been tested yet.

Hope this helps!

**Comment by inhahe** [comment]
Are you saying that if I download the latest version of GVDB I shouldn't
have this problem with CUDPP? Because I downloaded it just a few days ago,
version 1.1 (not sure if it's 1.1.1). Where should I download the latest
version from?

**Comment by NeilBickford-NV** [comment]
Yep, hopefully! (Oh, I see now - the NVIDIA website still has version 1.1, which has CUDPP! Let me see if I can get that updated.)

You can get the latest version by cloning this repository into your directory of choice (or from the Clone or download button on the main GitHub page!) The directory structure should be similar to the `source/` directory of this repository, but there should be no `shared_cudpp` folder in the `source` directory.

**Comment by inhahe** [comment]
Thanks. What version of Visual Studio should I use? Should I pass it to
cmake via 'cmake -G "Visual Studio 15 2017 Win64' or whatever version? And
should I specify the CUDA version? I have many versions of CUDA installed.
Should I do it via 'cmake -D CUDA_TOOLKIT_ROOT_DIR="C:/Program Files/NVIDIA
GPU Computing Toolkit/CUDA/v9.1"'?

**Comment by inhahe** [comment]
I was unable to build GVDB. I got the error:

CMake Error at C:/Program
Files/CMake/share/cmake-3.17/Modules/CMakeDetermineCompilerId.cmake:367
(message):
  No CUDA toolset found.
Call Stack (most recent call first):
  C:/Program
Files/CMake/share/cmake-3.17/Modules/CMakeDetermineCompilerId.cmake:32
(CMAKE_DETERMINE_COMPILER_ID_BUILD)
  C:/Program
Files/CMake/share/cmake-3.17/Modules/CMakeDetermineCUDACompiler.cmake:72
(CMAKE_DETERMINE_COMPILER_ID)
  source/gvdb_library/CMakeLists.txt:10 (project)

Even though I have the following CUDA versions installed:

 Directory of C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA

05/28/2020  08:57 PM    <DIR>          .
05/28/2020  08:57 PM    <DIR>          ..
05/12/2020  08:27 PM    <DIR>          v10.1
05/12/2020  06:04 PM    <DIR>          v10.2
05/24/2020  04:54 AM    <DIR>          v7.0
05/26/2020  06:34 PM    <DIR>          v7.5
05/28/2020  08:58 PM    <DIR>          v8.0
05/23/2020  11:31 PM    <DIR>          v9.1

I tried 'cmake -D CUDA_TOOLKIT_ROOT_DIR="C:/Program Files/NVIDIA GPU
Computing Toolkit/CUDA/v10.2" CMakeLists.txt' and I still got the error.
halp:P

**Comment by NeilBickford-NV** [comment]
Hmm, does it happen to work if you don't specify the `CUDA_TOOLKIT_ROOT_DIR`? It sounds like that's failing inside CMake, on line 10 of `gvdb_library\CMakeLists.txt`, `project(gvdb LANGUAGES CUDA CXX C)`. Specifying 'cmake -G "Visual Studio 15 2017 Win64"` sounds like the way to go - the build has been tested on this configuration, but we only support 64-bit configurations (and CMake's default for Visual Studio 2017 was x86).

**Comment by inhahe** [comment]
I had tried it both with and without specifying the CUDA root dir. I got
the error both times. I also tried the command you gave me now. I still got
the error.

------

C:\gvdb>cmake -G "Visual Studio 15 2017 Win64"
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


-- The C compiler identification is MSVC 19.16.27040.0
-- The CXX compiler identification is MSVC 19.16.27040.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual
Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual
Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -
works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual
Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual
Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -
works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at C:/Program
Files/CMake/share/cmake-3.17/Modules/CMakeDetermineCompilerId.cmake:367
(message):
  No CUDA toolset found.
Call Stack (most recent call first):
  C:/Program
Files/CMake/share/cmake-3.17/Modules/CMakeDetermineCompilerId.cmake:32
(CMAKE_DETERMINE_COMPILER_ID_BUILD)
  C:/Program
Files/CMake/share/cmake-3.17/Modules/CMakeDetermineCUDACompiler.cmake:72
(CMAKE_DETERMINE_COMPILER_ID)
  source/gvdb_library/CMakeLists.txt:10 (project)


-- Configuring incomplete, errors occurred!
See also "C:/gvdb/CMakeFiles/CMakeOutput.log".

------

I'll attach CMakeOutput.log in case you want it.
